services/src/main/scala/cromwell/pipeline/service/ProjectService.scala

```diff
@@ -34,11 +34,36 @@ class ProjectService(projectRepository: ProjectRepository)(implicit executionCon
     projectRepository.addProject(project)
   }
 
-  def deactivateProjectById(projectId: ProjectId): Future[Option[Project]] =
-    for {
-      _ <- projectRepository.deactivateProjectById(projectId)
-      getProject <- projectRepository.getProjectById(projectId)
-    } yield getProject
+  def deactivateProjectById(projectId: ProjectId, userId: UserId): Future[Option[Project]] = {
+    val result = projectRepository.getProjectById(projectId)
+    result.flatMap {
+      case Some(project) if project.ownerId != userId => throw new ProjectAccessDeniedException
+      case Some(_) =>
+        for {
+          _ <- projectRepository.deactivateProjectById(projectId)
+          getProject <- projectRepository.getProjectById(projectId)
+        } yield getProject
+      case None => throw new ProjectNotFoundException
+    }
+  }
+
+  def updateProject(request: ProjectUpdateRequest, userId: UserId): Future[Int] =
+    projectRepository
+      .getProjectById(request.projectId)
+      .flatMap(
+        projectOpt =>
+          projectOpt.map(
+            project =>
+              if (project.ownerId == userId)
+                projectRepository.updateProject(
+                  project.copy(name = request.name, repository = request.repository)
+                )
+              else Future.failed(new ProjectAccessDeniedException)
+          ) match {
+            case Some(value) => value
+            case None        => Future.failed(new ProjectNotFoundException)
+          }
```

Comment:
It is really hard to read this code. I believe it can be simplified by using pattern matching on the `projectOpt`.

Reply:
fixed

Reply:
Nothing has changed. Therefore, nothing is fixed.

Reply:
Sorry, I didn't push. Fixed now